src/test/java/ca/nexapp/core/persistence/unitofwork/UnitOfWorkSuiteITest.java

```diff
@@ -0,0 +1,127 @@
+package ca.nexapp.core.persistence.unitofwork;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import ca.nexapp.core.persistence.unitofwork.exceptions.AlreadyInTransactionException;
+import ca.nexapp.core.persistence.unitofwork.exceptions.NotInTransactionException;
+
+public abstract class UnitOfWorkSuiteITest<T extends AutoCloseable> {
```

Comment:
It is not a `Suite`. I would simply call it `UnitOfWorkTest`. The implementation will indicate if it is a unit or integration test.